code/userprofile/build.gradle.kts

```diff
@@ -0,0 +1,358 @@
+        }
+
+        exclude(BuildUtils.BUILD_CONFIG_CLASS, BuildUtils.R_CLASS)
+        options {
+            memberLevel = JavadocMemberLevel.PUBLIC
+        }
+    }
+}
+
+tasks.register<Javadoc>("javadoc") {
+    options.memberLevel = JavadocMemberLevel.PUBLIC
+}
+
+tasks.register<Jar>("javadocPublish") {
+    from(tasks.getByName<Javadoc>("javadoc"))
+    archiveClassifier.set("javadoc")
+}
+
+tasks.named("publish").configure {
+    dependsOn(tasks.named("assemblePhone"))
+}
+
+configure<PublishingExtension> {
+    publications {
+        create<MavenPublication>("release") {
+            groupId = buildUtils.groupId
+            artifactId = buildUtils.moduleName
+            version = buildUtils.versionToUse
+
+            artifact("$buildDir/outputs/aar/${buildUtils.moduleAARName}")
+            artifact(tasks.getByName<Jar>("javadocPublish"))
+
+            pom {
+                name.set(buildUtils.mavenRepoName)
+                description.set(buildUtils.mavenRepoDescription)
+                url.set(BuildUtils.DEVELOPER_DOC_URL)
+
+                licenses {
+                    license {
+                        name.set(BuildUtils.LICENSE_NAME)
+                        url.set(BuildUtils.LICENSE_URL)
+                        distribution.set(BuildUtils.LICENSE_DIST)
+                    }
+                }
+
+                developers {
+                    developer {
+                        id.set(BuildUtils.DEVELOPER_ID)
+                        name.set(BuildUtils.DEVELOPER_NAME)
+                        email.set(BuildUtils.DEVELOPER_EMAIL)
+                    }
+                }
+
+                scm {
+                    connection.set(BuildUtils.SCM_CONNECTION_URL)
+                    developerConnection.set(BuildUtils.SCM_CONNECTION_URL)
+                    url.set(BuildUtils.SCM_REPO_URL)
+                }
+
+                withXml {
+                    val dependenciesNode = asNode().appendNode("dependencies")
+
+                    val coreDependencyNode = dependenciesNode.appendNode("dependency")
+                    coreDependencyNode.appendNode("groupId", "com.adobe.marketing.mobile")
+                    coreDependencyNode.appendNode("artifactId", "core")
+                    coreDependencyNode.appendNode("version", buildUtils.mavenCoreVersion)
+                }
+            }
+        }
+    }
+
+    repositories {
+        maven {
+            name = "sonatype"
+            url = uri(buildUtils.getPublishUrl())
+            credentials {
+                username = System.getenv("SONATYPE_USERNAME")
+                password = System.getenv("SONATYPE_PASSWORD")
+            }
+        }
+    }
+}
+
+signing {
+    useGpgCmd()
+
+    setRequired(
+        tasks.withType<PublishToMavenRepository>().find {
+            gradle.taskGraph.hasTask(it)
+        }
+    )
+
+    sign(publishing.publications)
+}
+
+class BuildUtils {
```

Comment:
I really like moving all the variables in one location.. It makes it easier to share this code. However, I'm a little concerned with the different places we set variables, BuildUtils, gradle.properties, root project extras. Can this class be defined some place else, such as gradle.properties or root build.gradle?

Reply:
Agree! I was actually trying to get this moved a common location like `buildSrc` so that these contents can be accessed across the project (like mentioned [here](https://proandroiddev.com/better-dependencies-management-using-buildsrc-kotlin-dsl-eda31cdb81bf)) but was facing issues with this method. I will look at it once more this week and update my findings on this.

Reply:
@kevinlind , @praveek - I updated the PR to use `buildSrc` approach for managing constants. 
However, I left out  the constants in `gradle.properties` - `moduleVersion`, `moduleName`, `mavenRepoName`, `moduleAARName`, `mavenRepoDescription`, `mavenCoreVersion`. What are your thoughts on moving them also into BuildConstants?

Reply:
Discussed and agreed with @praveek that we will change this later as it impacts version upgrade scripts.